pkg/gktest/result.go

```diff
@@ -1,14 +1,45 @@
 package gktest
 
-// Result is the structured form of the results of running a test.
-type Result struct{}
+// SuiteResult is the Result of running a Suite of tests.
+type SuiteResult struct {
+	// Path is the absolute path to the file which defines Suite.
+	Path string
 
-func (r Result) String() string {
-	return ""
+	// Error is the error which stopped the Suite from executing.
+	// If defined, Results is empty.
+	Error error
+
+	// Results are the results of running the tests for each defined
+	// Template/Constraint pair.
+	Results []ConstraintResult
+}
+
+// ConstraintResult is the results of:
+// 1) Compiling the ConstraintTemplate,
+// 2) Instantiating the Constraint, and
+// 3) Running all Tests defined for the Constraint.
+type ConstraintResult struct {
```

Comment:
I think we need to figure out the new terminology in the design doc before moving forward here.

Reply:
Let's discuss - I've set up a meeting.

Reply:
Resolved by replacing Suite/Constraint/Test with Suite/Test/Case.